React/prop-types.md

```diff
@@ -0,0 +1,82 @@
+# Prop Types
+
+## Learning Goals
+By the end of this lesson you should be able to:
+
+- Explain the use of `PropTypes`
+- Use `PropTypes` to define and document the props a component can take
+
+You may have noticed this message from your JavaScript linter.  "____ is missing from props validation (react/prop-types)."  The linter is encouraging you to use a feature of React called Prop-types in which you list the props a component takes and their types.
+
+![prop types error](images/prop-types.png)
+
+In this lesson we will define `propTypes` in our React components which will provide some type-checking and documentation of our React component props.
+
+## Student Component Example
+
+Our Student component takes a name & an e-mail as props.  We can define the properties our `Student` component takes by adding a `static` `propTypes` attribute.
+
+First you need to `import PropTypes from 'prop-types';` to import the package.  Then add a static `propTypes` attribute to your class like below.
+
+```javascript
+
+static propTypes = {
+  name: PropTypes.string.isRequired,
+  email: PropTypes.string.isRequired,
+}
+```
+
+The above block defines two props.  They both must be Strings and are required to use a `Student` component.  If you try to use the `Student` component without a `name` prop you will get this error message:
+
+![Error Msg](images/missing-prop.png)
+
+If a component has optional props you can simply leave off the `isRequired` attribute.
+
+**Exercise** Add another prop to the `Student` component called `age`.  This component should be _optional_.  Update the `Student` class' `propTypes`.
```

Comment:
You could also add the "present" attribute which we're going to use in the creating components lecture as optional boolean?

Reply:
Added